File: Languages/Fsharp/RandAvg.fs
open System
open System.IO
open System.Collections.Generic

// Hello, World!
let hello () = printfn "Hello, World!"

let generateRandomNumbers () =
    let count = 1000
    let outputDir = "../rand_avg output"
    let outputFile = outputDir + "/random_numbers.txt"
    
    // Create output directory
    if not (Directory.Exists(outputDir)) then
        Directory.CreateDirectory(outputDir) |> ignore
    
    // Random number generator
    let random = Random()
    
    // Generate 1000 random numbers
    let randomNumbers = List<int>()
    let mutable sum = 0L
    
    for i in 1..count do
        let num = random.Next(0, 1000)
        randomNumbers.Add(num)
        sum <- sum + int64 num
    
    // Calculate mean
    let mean = float sum / float count
    
    // Save to file
    try
        use writer = new StreamWriter(outputFile)
        for num in randomNumbers do
            writer.WriteLine(num)
        
        printfn "Generated 1000 random numbers"
        printfn "Mean: %.2f" mean
        printfn "Saved to: %s" outputFile
    with
    | ex -> printfn "Could not create output file"

[<EntryPoint>]
let main argv =
    hello()
    generateRandomNumbers()
    0

File: Languages/Csharp/RandAvg.cs
using System;
using System.IO;
using System.Collections.Generic;

public class RandAvg
{
    public static void Main(string[] args)
    {
        // Hello, World!
        Console.WriteLine("Hello, World!");
        GenerateRandomNumbers();
    }
    
    private static void GenerateRandomNumbers()
    {
        const int count = 1000;
        const string outputDir = "../rand_avg output";
        const string outputFile = outputDir + "/random_numbers.txt";
        
        // Create output directory
        if (!Directory.Exists(outputDir))
        {
            Directory.CreateDirectory(outputDir);
        }
        
        // Random number generator
        Random random = new Random();
        
        // Generate 1000 random numbers
        List<int> randomNumbers = new List<int>();
        long sum = 0;
        
        for (int i = 0; i < count; i++)
        {
            int num = random.Next(0, 1000);
            randomNumbers.Add(num);
            sum += num;
        }
        
        // Calculate mean
        double mean = (double)sum / count;
        
        // Save to file
        try
        {
            using (StreamWriter writer = new StreamWriter(outputFile))
            {
                foreach (int num in randomNumbers)
                {
                    writer.WriteLine(num);
                }
            }
            
            Console.WriteLine("Generated 1000 random numbers");
            Console.WriteLine($"Mean: {mean:F2}");
            Console.WriteLine($"Saved to: {outputFile}");
        }
        catch (Exception e)
        {
            Console.WriteLine("Could not create output file");
        }
    }
}
